### PR TITLE
[YUNIKORN-979] Add e2e test coverage for the admission controller

### PR DESCRIPTION
**Pull request by lowc1012**
### What is this PR for?
Add the e2e test for the admission controller to increase test coverage.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
[YUNIKORN-979](https://issues.apache.org/jira/projects/YUNIKORN/issues/YUNIKORN-979)

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Comment by wilfred-s** [comment]
@lowc1012 can you have a look at the two lint issues?
The kClient is a strange one to get. Not something you would expect. It is linked to not using the Hungarian notation. The go lint code specifically picks up on the `k` which stands for kind in that case. Rename the variable to kubeClient or just to client can be used.

**Comment by wilfred-s** [comment]
One other thing: should we check that the admission controller is up and running by calling the health check before we run these tests or is that called earlier in the e2e tests?

**Comment by lowc1012** [comment]
@wilfred-s Thanks for your explanation.
About the second comment, Do we have the health check API endpoint for admission controller? If not, should I implement it in the PR or create another JIRA for that? Thank you!

**Comment by lowc1012** [comment]
I have no idea why the test for namespaceBlacklist failed.
In the test, I created a pod in kube-system namespace, it should not be mutated by admission controller. 
[YUNIKORN-1084](https://issues.apache.org/jira/projects/YUNIKORN/issues/YUNIKORN-1084) may address this problem.

**Comment by wilfred-s** [comment]
@craigcondit can you please check if the e2e test failed as expected and that this will work in combination with your changes. If we need changes for YUNIKORN-1084 I think we need to commit this first and add changes to that PR.

**Comment by craigcondit** [comment]
> @craigcondit can you please check if the e2e test failed as expected and that this will work in combination with your changes. If we need changes for YUNIKORN-1084 I think we need to commit this first and add changes to that PR.

Yes, this test will fail currently because the changes from YUNIKORN-1084 have not yet been merged (the existing behavior is not correct, but this test assumes it is). I don't think we need changes to this, but YUNIKORN-1084 needs to go in first to get the admission controller to exhibit the correct behavior.

**Comment by craigcondit** [comment]
@lowc1012 can you re-base this on current master? I think all the admission controller fixes are in now, so hopefully the tests pass.

File: test/e2e/admission_controller/admission_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		err = kubeClient.WaitForPodBySelectorRunning(ns,
+			fmt.Sprintf("app=%s", sleepRespPod.ObjectMeta.Labels["app"]), 10)
+		gomega.Ω(err).ShouldNot(gomega.HaveOccurred())
+		gomega.Ω(sleepRespPod.Spec.SchedulerName).Should(gomega.BeEquivalentTo(constants.SchedulerName))
+	})
+
+	ginkgo.It("Verifying a pod is created on namespace blacklist", func() {
+		ginkgo.By("Create a pod in namespace blacklist")
+		podConfigs := common.SleepPodConfig{Name: sleepPodName, NS: blackNs}
+		pod, err := kubeClient.CreatePod(common.InitSleepPod(podConfigs), blackNs)
```
Comment (craigcondit) [inline test/e2e/admission_controller/admission_controller_test.go:60]: Using `InitSleepPod()` here will explicitly set `schedulerName`: `yunikorn`. This prevents the test from passing.

**Review by craigcondit — CHANGES_REQUESTED** [review]

File: test/e2e/admission_controller/admission_controller_test.go (hunk 1 above)
Comment (lowc1012) [reply test/e2e/admission_controller/admission_controller_test.go]: I see and am sorry for my mistake.
Thank you.

**Review by craigcondit — APPROVED** [review]
+1 LGTM.

File: test/e2e/admission_controller/admission_controller_test.go (hunk 1 above)
Comment (craigcondit) [reply test/e2e/admission_controller/admission_controller_test.go]: No worries! Thanks for the contribution.

**Pull request closed without merging**